packages/core/test/waitFor.test.ts

```diff
@@ -48,4 +48,45 @@ describe('waitFor', () => {
       expect(e).toBeInstanceOf(Error);
     }
   });
+
+  it('should resolve correctly when the predicate immediately matches the current state', async () => {
+    const machine = createMachine({
+      initial: 'a',
+      states: {
+        a: {}
+      }
+    });
+
+    const service = interpret(machine).start();
+
+    await expect(
+      waitFor(service, (state) => state.matches('a'))
+    ).resolves.toHaveProperty('value', 'a');
```

Comment:
this was throwing before the fix since internal `sub` wasn't yet returned when the predicate matched so it was just crashing on `sub.unsubscribe()`